docs/did/asset-did-spec.md

```diff
@@ -0,0 +1,173 @@
+
+```
+asset-did                   = "did:asset:" + <modified-caip-19-asset-id>
+modified-caip-19-asset-id   = <caip-2-chain-id> + "." + <asset-namespace> + ":" + <asset-reference> + [":" + <asset-id>]
+caip-2-chain-id             = <chain-namespace> + ":" + <chain-reference>
+chain-namespace             = [-a-z0-9]{3,8}
+chain-reference             = [-a-zA-Z0-9]{1,32}
+asset-namespace             = [-a-z0-9]{3,8}
+asset-reference             = [-a-zA-Z0-9]{1,64}
+asset-id                    = [-a-zA-Z0-9]{1,78}
+```
+
+The `caip-2-chain-id` is a CAIP (Chain Agnostic Improvement Proposal) chain identifier as described in the [CAIP-2 Blockchain ID Specification][caip-2-spec].
+The `modified-caip-19-asset-id` is derived from a CAIP asset identifier as described in the [CAIP-19 Asset Type and Asset ID Specification][caip-19-spec], with the `/` symbol separating the chain identifier from the asset identifier being replaced with the `.` symbol, and the optional `/` symbol preceding the `asset-id` component being replaced by the `:` symbol.
+This is required to render the resulting string a valid URI according to the [DID Core Specification][did-core-spec].
+
+Chain and asset namespaces are defined with Improvement Proposals and are stored under the [CAIP repository][caip-repo].
+For instance, the namespaces for [EIP155][caip-3-spec] (Ethereum-based) and [BIP122][caip-4-spec] (Bitcoin-based) chains as well as for [SLIP44][caip-20-spec] (native fungible tokens), [ERC20][caip-21-spec] (smart contract based tokens), and [ERC721][caip-22-spec] (NFTs) assets have already been defined.
+
+## Method Operations
+
+### Create an Asset DID
+
+An Asset DID as such is not directly created, but it is generated at any time from a specific asset on as specific network.
+Because of the chain agnostic nature of Asset DIDs, this process can take place on any blockchain that is identifiable by a [CAIP-2 Identifier][caip-2-spec].
+
+Furthermore, as far as Asset DIDs are concerned, the assets identified do not need to already exist.
+For instance, a "right to create" a new NFT collection on a given blockchain for a user could be encoded in a [Verifiable Credential][vc-spec] where the identifier of the asset is an Asset DID.
+
+### Resolve an Asset DID
+
+The information required to resolve an Asset DID is the following:
+
+- The **chain** on which the asset lives, e.g., the Ethereum mainnet
+- The **namespace** defining the asset, e.g., the ERC721 namespace
+- The **reference** to the asset class within the defined namespace, e.g., the smart contract address for an NFT collection
+- [OPTIONAL] The unique **identifier** of the asset instance within the asset class, e.g., the NFT ID within a given collection
+
+The resolution process then follows a hierarchical structure.
+
+The chain identifier is decomposed into its namespace and reference.
+Then, the resolver for the specified namespace is used to resolve the specified reference.
```

Comment:
Is this just how it works on-chain in our codebase?
I would've expected, that we just have a map of `Asset DID -> DID Document`, but apparently we store it in a more complex way, is that correct?

If the word `resolve` here references to querying the blockchain, i think we should say that explicitly.

Reply:
No this has nothing to do with the chain. The chain (now) simply stores a map of (asset DID + credential hash) -> block number information. The asset DID is parsed from a UTF8-encoded string. This piece of information is not relevant anymore if we scrape any chain-specific resolution logic from the resolver tho. So we might want to discuss about this as well.

Reply:
No this has nothing to do with the chain. The chain (now) simply stores a map of (asset DID + credential hash) -> block number information. The asset DID is parsed from a UTF8-encoded string. This piece of information is not relevant anymore if we scrape any chain-specific resolution logic from the resolver tho. So we might want to discuss about this as well.